app/models/test_track/web_session.rb

```diff
@@ -93,7 +93,7 @@ def cookie_domain
   end
 
   def _cookie_domain
-    if bare_ip_address? || fully_qualified_cookie_domain_enabled?
+    if bare_ip_address? || request.local? || fully_qualified_cookie_domain_enabled?
```

Comment:
does this have weird interactions with consider_all_requests_local or anything?

Reply:
nope: https://github.com/rails/rails/blob/98a57aa5f610bc66af31af409c72173cdeeb3c9e/railties/lib/rails/application_controller.rb#L21-L23